### PR TITLE
feat(config): add parameters 12, 50, 51 and 52 for Inovelli LZW31-SN

### PR DESCRIPTION
**Pull request by otaku**
Sourced from https://github.com/InovelliUSA/SmartThingsInovelli/blob/b9fb2ad351b693a2a0bffb85debd98f27979fe44/devicetypes/inovelliusa/inovelli-dimmer-red-series-lzw31-sn.src/inovelli-dimmer-red-series-lzw31-sn.groovy

### CODE REVIEW

**Comment by DelusionalAI** [comment]
Inovelli doesn't have that same smart-things config that you referenced updated for the LZW31, but they should share all the same parameters that that LZW31-SN has, even down the same firmware version numbers. 

https://support.inovelli.com/portal/en/kb/articles/firmware-change-log-lzw31-dimmer-switch-black-series

Should that be added here as well or a different PR?

**Comment by blhoward2** [comment]
Never mind. It looks like Innovelli submitted it.

**Comment by otaku** [comment]
I can copy the changes over to the LZW31 as well if you'd like it to be in this PR.

Only other differences I saw from their SmartThings config was preset values for the LED indicator colors

**Comment by blhoward2** [comment]
Sure

**Comment by AlCalzone** [comment]
Just request a review once you are done with it 👍🏻

**Comment by otaku** [comment]
Requesting a review!

File: packages/config/config/devices/0x031e/lzw31-bsd_1.47-1.47.json [hunk 1]
```diff
@@ -0,0 +1,393 @@
...
+			"label": "Energy Reports",
+			"unit": "%",
+			"valueSize": 1,
+			"minValue": 0,
+			"maxValue": 100,
+			"defaultValue": 10,
+			"readOnly": false,
+			"writeOnly": false,
+			"allowManualEntry": true
+		},
```
Comment (kelchm) [inline packages/config/config/devices/0x031e/lzw31-bsd_1.47-1.47.json]: It looks like this was propagated from the existing config file, but the LZW31 (Black Series Dimmer) does not support led strip effects (notifications) or energy reporting.

File: packages/config/config/devices/0x031e/lzw31-bsd_0.0-1.45.json [hunk 2]
```diff
@@ -146,6 +146,78 @@
...
+		"12": {
+			"label": "Association Behavior",
+			"valueSize": 1,
+			"minValue": 0,
+			"maxValue": 15,
+			"defaultValue": 15,
+			"readOnly": false,
+			"writeOnly": false,
+			"allowManualEntry": false,
+			"options": [
```
Comment (AlCalzone) [inline packages/config/config/devices/0x031e/lzw31-bsd_0.0-1.45.json]: These options are best modeled as partial parameters. I think we have that exact same param in another LZW config

File: packages/config/config/devices/0x031e/lzw31-bsd_1.47-1.47.json [hunk 3]
```diff
@@ -0,0 +1,393 @@
...
+		"12": {
+			"label": "Association Behavior",
+			"valueSize": 1,
+			"minValue": 0,
+			"maxValue": 15,
+			"defaultValue": 15,
+			"readOnly": false,
+			"writeOnly": false,
+			"allowManualEntry": false,
+			"options": [
```
Comment (AlCalzone) [inline packages/config/config/devices/0x031e/lzw31-bsd_1.47-1.47.json]: dito

File: packages/config/config/devices/0x031e/lzw31-bsd_1.49.json [hunk 4]
```diff
@@ -0,0 +1,404 @@
...
+		"12": {
+			"label": "Association Behavior",
+			"valueSize": 1,
+			"minValue": 0,
+			"maxValue": 15,
+			"defaultValue": 15,
+			"readOnly": false,
+			"writeOnly": false,
+			"allowManualEntry": false,
+			"options": [
```
Comment (AlCalzone) [inline packages/config/config/devices/0x031e/lzw31-bsd_1.49.json]: dito

File: packages/config/config/devices/0x031e/lzw31-sn_0.0-1.45.json [hunk 5]
```diff
@@ -146,6 +146,78 @@
...
+		"12": {
+			"label": "Association Behavior",
+			"valueSize": 1,
+			"minValue": 0,
+			"maxValue": 15,
+			"defaultValue": 15,
+			"readOnly": false,
+			"writeOnly": false,
+			"allowManualEntry": false,
+			"options": [
```
Comment (AlCalzone) [inline packages/config/config/devices/0x031e/lzw31-sn_0.0-1.45.json]: dito

File: packages/config/config/devices/0x031e/lzw31-sn_1.47-1.47.json [hunk 6]
```diff
@@ -0,0 +1,393 @@
...
+		"12": {
+			"label": "Association Behavior",
+			"valueSize": 1,
+			"minValue": 0,
+			"maxValue": 15,
+			"defaultValue": 15,
+			"readOnly": false,
+			"writeOnly": false,
+			"allowManualEntry": false,
+			"options": [
```
Comment (AlCalzone) [inline packages/config/config/devices/0x031e/lzw31-sn_1.47-1.47.json]: dito

File: packages/config/config/devices/0x031e/lzw31-sn_1.49.json [hunk 7]
```diff
@@ -0,0 +1,404 @@
...
+		"12": {
+			"label": "Association Behavior",
+			"valueSize": 1,
+			"minValue": 0,
+			"maxValue": 15,
+			"defaultValue": 15,
+			"readOnly": false,
+			"writeOnly": false,
+			"allowManualEntry": false,
+			"options": [
```
Comment (AlCalzone) [inline packages/config/config/devices/0x031e/lzw31-sn_1.49.json]: dito

**Review by AlCalzone — CHANGES_REQUESTED** [review]
Haven't checked the params if they are correct. Mostly looks good, except for that one parameter

**Comment by blhoward2** [comment]
You can find an example of the partial parameter in packages/config/config/devices/0x0312/ms10z.json

File: packages/config/config/devices/0x031e/lzw31-bsd_0.0-1.45.json (hunk 2 above)
Comment (Hilo4321) [reply packages/config/config/devices/0x031e/lzw31-bsd_0.0-1.45.json]: I don't see that this parameter exists in the [manual](https://support.inovelli.com/portal/api/kbArticles/407409000001873356/attachments/1abnf924b37dde28f4ea1bba3d1a3911d7384/content?portalId=1ce1db55e4825f16146555b24f0f1753cbf5f7d83ca0a3d8b31013b2a2d69e1f&inline=true) or in the [firmware history](https://support.inovelli.com/portal/en/kb/articles/firmware-change-log-lzw31-dimmer-switch-black-series). Am I missing something?

Comment (DelusionalAI) [reply packages/config/config/devices/0x031e/lzw31-bsd_0.0-1.45.json]: They dont have that one in the manual for some reason, but it is a valid parameter and you can find it in their OZW [config files](https://github.com/InovelliUSA/OpenZWave/blob/master/config/inovelli/lzw31-sn.xml) and their [SmartThings device handlers](https://github.com/InovelliUSA/OpenZWave/blob/master/config/inovelli/lzw31-sn.xml) right from Inovelli themselves.

Comment (blhoward2) [reply packages/config/config/devices/0x031e/lzw31-bsd_0.0-1.45.json]: The SmartThings driver linked was provided by Inovelli to SmartThings. They appear to have undocumented parameters available.

File: packages/config/config/devices/0x031e/lzw31-bsd_1.47-1.47.json (hunk 1 above)
Comment (otaku) [reply packages/config/config/devices/0x031e/lzw31-bsd_1.47-1.47.json]: https://support.inovelli.com/portal/en/kb/articles/products-switches-dimmer-lzw31-spec-sheet

I noticed it has parameters 17 in their spec sheet.

**Comment by otaku** [comment]
I accidently closed this pull request somehow, can it be reopened?  Changes to partial parameters were done and removed the parameters in LZW31 BSD as requested.

**Comment by blhoward2** [comment]
It won't let me do it. @AlCalzone ?

**Comment by AlCalzone** [comment]
Reopened it. Usually the author should be able to reopen though.

**Comment by AlCalzone** [comment]
Hmm doesn't work. Did you remove all changes compared to my master branch?

**Comment by otaku** [comment]
I'll just re-open a new PR... I had too many branches and pushed the wrong one to my own.

**Comment by blhoward2** [comment]
Glad it wasn't just me. I have a button to reopen it and it initially said it did, but then it goes right back to closed.

**Pull request closed without merging**